worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/XmlConverter.java

```diff
@@ -123,6 +145,89 @@ public static Document createXmlDocument(final List<XmlFieldEntry> xmlFieldEntri
         return doc;
     }
 
+    private static void addEmailHeadersToXmlFieldEntries(String fieldToAddTo, List<XmlFieldEntry> sourceEntries)
+    {
+        StringBuilder headersBuilder = new StringBuilder();
+        // Build headers
+        appendEmailHeaderToBuilder(headersBuilder, "From", "from", sourceEntries);
+        appendEmailHeaderToBuilder(headersBuilder, "To", "to", sourceEntries);
+        appendEmailHeaderToBuilder(headersBuilder, "CC", "cc", sourceEntries);
+        appendEmailHeaderToBuilder(headersBuilder, "BCC", "bcc", sourceEntries);
+        appendEmailDateHeaderToBuilder(headersBuilder, sourceEntries);
+        appendEmailHeaderToBuilder(headersBuilder, "Subject", "subject", sourceEntries);
+
+        // Add a new line indicating break in headers section.
+        headersBuilder.append("\n");
+
+        for(XmlFieldEntry entryToUpdate: sourceEntries.stream()
+                .filter(ent -> ent.getName().equals(fieldToAddTo))
+                .collect(Collectors.toList())){
+            entryToUpdate.setText(headersBuilder.toString() + entryToUpdate.getText());
+        }
+    }
+
+    private static void appendEmailDateHeaderToBuilder(StringBuilder headersBuilder, List<XmlFieldEntry> sourceEntries){
```

Comment:
I don't get why you need all that logic... Headers get normalized in a different place... The only thing in the content should be values of fields... Simple array with field names to copy...

Reply:
It was to duplicate the same behavior as text extract for outputting the header value. Today the only value used for the sent date is the UTC attribute on the element so we don't need to control the format right now.